packages/schema/compose/src/index.ts

```diff
@@ -34,24 +32,26 @@ export async function composeSchema(
   options: ComposerOptions
 ): Promise<ComposerOutput> {
   const { schemas, resolvers } = options;
-  const typeInfos: {
-    [name: string]: TypeInfo;
-  } = {};
+  const typeInfos: Partial<Record<SchemaKind, TypeInfo>> = {};
 
   if (Object.keys(schemas).length === 0) {
     throw Error("No schema provided");
   }
 
-  for (const name of Object.keys(schemas)) {
-    const schema = schemas[name];
+  for (const kind of schemaKinds) {
+    const schema = schemas[kind];
+
+    if (schema === undefined) {
+      continue;
+    }
```

Comment:
Why don't use `Object.entries(schemas)` here?

Reply:
It's because we require `kind` to be fully typed in this codeblock as `SchemaKind`. Doing Object.keys(...) or Object.values(...) removes this typing as it's too generic for our purpose here.